pkg/crc/errors/atexit.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright (C) 2019 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"os"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/output"
+)
+
+const ExitHandlerPanicMessage = "At least one exit handler vetoed to exit program execution"
+
+// exitHandlers keeps track of the list of registered exit handlers. Handlers are applied in the order defined in this list.
+var exitHandlers = []func(code int) bool{}
```

Comment:
Can we maybe have a type for `exitHandler` :)

Reply:
Not following you...

Reply:
`type exitHandler func(int) bool` and use this, mostly because of readability..

Reply:
no... these are Handlers, multiple. this is a list of possible functions that get triggered upon request. that is why in code you find `range`.

Reply:
@anjannath did that work for you?

Reply:
@gbraad Yup.. 
```diff
diff --git a/pkg/crc/errors/atexit.go b/pkg/crc/errors/atexit.go
index c59d4a9..a9d1784 100644
--- a/pkg/crc/errors/atexit.go
+++ b/pkg/crc/errors/atexit.go
@@ -25,8 +25,10 @@ import (
 
 const ExitHandlerPanicMessage = "At least one exit handler vetoed to exit program execution"
 
+type exitHandlerFunc func(int) bool
+
 // exitHandlers keeps track of the list of registered exit handlers. Handlers are applied in the order defined in this list.
-var exitHandlers = []func(code int) bool{}
+var exitHandlers = []exitHandlerFunc{}
 
 // Exit runs all registered exit handlers and then exits the program with the specified exit code using os.Exit.
 func Exit(code int) {
@@ -49,13 +51,13 @@ func ExitWithMessage(code int, msg string) {
 }
 
 // Register registers an exit handler function which is run when Exit is called
-func RegisterExitHandler(exitHandler func(code int) bool) {
-	exitHandlers = append(exitHandlers, exitHandler)
+func RegisterExitHandler(handler exitHandlerFunc) {
+	exitHandlers = append(exitHandlers, handler)
 }
 
 // ClearExitHandler clears all registered exit handlers
 func ClearExitHandler() {
-	exitHandlers = []func(code int) bool{}
+	exitHandlers = []exitHandlerFunc{}
 }
 
 // runHandlers runs all registered exit handlers, passing on the intended exit code.
@@ -70,7 +72,7 @@ func runHandlers(code int) bool {
 }
 
 // runHandler runs the single specified exit handler, returning whether this handler vetos the exit or not.
-func runHandler(exitHandler func(code int) bool, code int) bool {
+func runHandler(handler exitHandlerFunc, code int) bool {
 	defer func() {
 		err := recover()
 		if err != nil {
@@ -78,5 +80,5 @@ func runHandler(exitHandler func(code int) bool, code int) bool {
 		}
 	}()
 
-	return exitHandler(code)
+	return handler(code)
 }
```

Reply:
fixing this in #63 ... can we move ahead with this and consider a refactor in followup commits?

Reply:
Sure..